dbt_style_guide.md

```diff
@@ -1,237 +1,415 @@
+  left join users as riders
     on trips.rider_id = riders.user_id
 
-```
+  ```
+
+- ### Example SQL
+  ```sql
+  with
+
+  my_data as (
+      select * from {{ ref('my_data') }}
+      where not is_deleted
+  ),
+
+  some_cte as (
+      select * from {{ ref('some_cte') }}
+  ),
+
+  some_cte_agg as (
+      select
+          id,
+          sum(field_4) as total_field_4,
+          max(field_5) as max_field_5
+      from some_cte
+      group by 1
+  ),
+
+  final as (
+      select [distinct]
+          my_data.field_1,
+          my_data.field_2,
+          my_data.field_3,
+
+          -- use line breaks to visually separate calculations into blocks
+          case
+              when my_data.cancellation_date is null
+                and my_data.expiration_date is not null
+                then expiration_data
+              when my_data.cancellation_date is null
+                then my_data.start_date + 7
+              else my_data.cancellation_date
+          end as cancellation_date,
+
+          some_cte_agg.total_field_4,
+          some_cte_agg.max_field_5
+      from my_data
+      left join some_cte_agg  
+          on my_data.id = some_cte_agg.id
+      where 
+        my_data.field_1 = 'abc'
+        and (
+            my_data.field_2 = 'def'
+            or my_data.field_2 = 'ghi'
+        )
+      having count(*) > 1
+  )
+
+  select * from final
+  ```
 
 ## YAML style guide
 
-* Indents should be two spaces
-* List items should be indented
-* Use a new line to separate list items that are dictionaries where appropriate
-* Lines of YAML should be no longer than 80 characters.
-
-### Example YAML
-```yaml
-version: 2
-
-models:
-  - name: events
-    columns:
-      - name: event_id
-        description: This is a unique identifier for the event
-        tests:
-          - unique
-          - not_null
-
-      - name: event_time
-        description: "When the event occurred in UTC (eg. 2018-01-01 12:00:00)"
-        tests:
-          - not_null
-
-      - name: user_id
-        description: The ID of the user who recorded the event
-        tests:
-          - not_null
-          - relationships:
-              to: ref('users')
-              field: id
-```
+- Every subdirectory contains their own `.yml` file(s) which contain configurations for the models within the subdirectory.
+
+- A separate `.yml` file is created per top-level configuration (`sources`, `models`) that applies to the models within the subdirectory. 
+
+- YAML files should be prefixed with an underscore to keep it at the top of the subdirectory.  
+  Example: `__sources_core.yml`, `_models_core.yml`
+
+- YAML files should be named for the top-level configuration it contains.  
+  Examples: `_sources_core.yml`, `_models_core.yml`
+
+- Indents should use four spaces.
+
+- List items should be indented.
+
+- Use a new line to separate list items that are dictionaries, where appropriate.
+
+- Lines of YAML should be no longer than 80 characters.

```

Comment:
I've always wondered, with how weird yml is about white space, is this something we recommend? I know in our description we can do ': > ' but if it isn't a field with that option can they create that white space without breaking something?

Reply:
I actually just tested this today with `accepted_values` and you can indeed do things like this and it will still work:
```
        tests:
          - accepted_values:
              values: [
                'AFRICA',
                'MIDDLE EAST',
                'ASIA','EUROPE','AMERICA']
```

As for descriptions, `:>` works, as does something like: 
```
version: 2

models:
  - name: dim_customers
    description: "
    This is a multi
    paragraph description of
    Customer dimensions table
    "
    columns:
      - name: customer_id
```

Reply:
Awesome, thanks Pat.